core/src/test/java/com/linecorp/armeria/client/ClientMaxConnectionAgeTest.java

```diff
@@ -160,4 +172,48 @@ void shouldCloseIdleConnectionByMaxConnectionAge(SessionProtocol protocol) {
             await().untilAtomic(closed, Matchers.is(1));
         }
     }
+
+    @EnumSource(value = SessionProtocol.class, names = "PROXY", mode = Mode.EXCLUDE)
+    @ParameterizedTest
+    void shouldCloseLongRequestsByMaxConnectionAge(SessionProtocol protocol) throws Exception {
+        try (ClientFactory factory = ClientFactory.builder()
+                                                  .connectionPoolListener(connectionPoolListener)
+                                                  .idleTimeoutMillis(0)
+                                                  .maxConnectionAgeMillis(MAX_CONNECTION_AGE)
+                                                  .tlsNoVerify()
+                                                  .build()) {
+            final WebClient client = WebClient.builder(server.uri(protocol))
+                                              .factory(factory)
+                                              .responseTimeoutMillis(0)
+                                              .build();
+
+            assertThat(client.get("/delayed?seconds=4").aggregate().join().status()).isEqualTo(OK);
```

Comment:
The request doesn't seem to be closed unlike what the method name says?

I maybe miss something, but if the `MAX_CONNECTION_AGE` is two seconds and the response is sent after 4 seconds, shouldn't the connection should be closed forcefully? 🤔

Reply:
Sorry about the late check

Maybe I'm misunderstanding the specification, but I thought that even if `MAX_CONNECTION_AGE` is reached, the client waits for requests to finish (so it doesn't forcefully fail long-running requests). I've updated the test name to better reflect this 😅 

1) A timer is scheduled for `MAX_CONNECTION_AGE`, but is only respected if there are no running requests
https://github.com/line/armeria/blob/b77bfbb931d8c5b51f65715db7a50c97c12ba622/core/src/main/java/com/linecorp/armeria/internal/common/AbstractKeepAliveHandler.java#L436-L440

2) Instead, every time a request may have finished, we check if `MAX_CONNECTION_AGE` has passed, and there are no running requests.

https://github.com/line/armeria/blob/b77bfbb931d8c5b51f65715db7a50c97c12ba622/core/src/main/java/com/linecorp/armeria/client/Http2ResponseDecoder.java#L153-L155
https://github.com/line/armeria/blob/b77bfbb931d8c5b51f65715db7a50c97c12ba622/core/src/main/java/com/linecorp/armeria/client/Http2ResponseDecoder.java#L215-L217
https://github.com/line/armeria/blob/b77bfbb931d8c5b51f65715db7a50c97c12ba622/core/src/main/java/com/linecorp/armeria/client/Http2ResponseDecoder.java#L275-L279

3) At the same time, we don't allow future requests for connections where `MAX_CONNECTION_AGE` has passed.
https://github.com/line/armeria/blob/b77bfbb931d8c5b51f65715db7a50c97c12ba622/core/src/main/java/com/linecorp/armeria/client/HttpSessionHandler.java#L163

Reply:
> I've updated the test name to better reflect this 😅

Thanks for that. 😄 

> but I thought that even if MAX_CONNECTION_AGE is reached, the client waits for requests to finish (so it doesn't forcefully fail long-running requests)

Exactly. My memory was wrong. Thanks for checking it. 😄